### PR TITLE
Changes to plugin loading functions

### PR DESCRIPTION
**Pull request by Henrybk**
New functions:
- getPluginsFiles
- getCondition
- unloadByRegexp
- unloadPlugins
- loadByRegexp
- loadPlugins
- reloadByRegexp
- reloadAll
- reloadPlugins

Also changed Commands.pm accordingly

### CODE REVIEW

**Comment by itsrachelfish** [comment]
@Henrybk Can you explain why these changes were made and how they should be tested? :+1:

**Comment by Henrybk** [comment]
I made these changes based on the files loading functions, they can be tested by just loading, unloading and reloading some plugins, I tested it and found no errors.

File: src/Plugins.pm [hunk 1]
```diff
@@ -68,16 +68,36 @@ use enum qw(CALLBACK USER_DATA);
...
+				push @files, getFilesFromDirs(["$dir/$item"], $f_exts, $d_ignores, $recurse_lv-1);
+			}
+		}
+	}
+	return @files;
+}
+
+sub getPluginsFiles {
+	my @folders = Settings::getPluginsFolders();
+	my @files = getFilesFromDirs(\@folders, 'pl|lp', 'cvs', 1);
```
Comment (itsrachelfish) [inline src/Plugins.pm]: What's a .lp file extension?

Comment (Henrybk) [inline src/Plugins.pm]: No idea, it was already there before my changes :P

**Comment by itsrachelfish** [comment]
Alright, hold the phone on this one. @Henrybk It looks like there's a bug in your code which prevents plugins from being loaded manually. Using the current code in the master branch, I can force load a plugin even if it's not enabled in sys.txt.

This is the current behavior:
```
plugin
-------------------------- Currently loaded plugins ---------------------------
#   Name                 Description
0   macro                allows usage of macros
1   profiles             Profiles Selector
2   breakTime            Automatically disconnect and reconnect at certain time
-------------------------------------------------------------------------------
plugin load plugins/buffplease.pl
Loading plugin plugins/buffplease.pl...
plugin
-------------------------- Currently loaded plugins ---------------------------
#   Name                 Description
0   macro                allows usage of macros
1   profiles             Profiles Selector
2   breakTime            Automatically disconnect and reconnect at certain time
3   Buff Please?         Buff people when they ask
-------------------------------------------------------------------------------
```

As you can see, the 3 plugins defined in sys.txt are loaded by default and I can load other plugins manually.

However, when I try to do the same thing using your branch, this happens:
```
plugin load plugins/buffplease.pl
Selectively loading plugins...
plugin
-------------------------- Currently loaded plugins ---------------------------
#   Name                 Description
0   macro                allows usage of macros
1   profiles             Profiles Selector
2   breakTime            Automatically disconnect and reconnect at certain time
-------------------------------------------------------------------------------
```

**Comment by kaliwanagan** [comment]
@Henrybk will you have time to address @itsrachelfish concerns?

**Comment by Henrybk** [comment]
Yeah, next week I'll be on vacation and will take a look at it.

**Comment by Nipodemos** [comment]
@Henrybk do you have free time to fix the error reported?

**Comment by ya4ept** [comment]
> @Henrybk do you have free time to fix the error reported?


I fixed it:

```
start.exe --no-connect

plugin
-------------------------- Currently loaded plugins ---------------------------
#   Name                 Description
0   macro                allows usage of macros
-------------------------------------------------------------------------------
plugin load event
Loading plugin plugins/eventMacro/eventMacro.pl...
plugin
-------------------------- Currently loaded plugins ---------------------------
#   Name                 Description
0   macro                allows usage of macros
1   eventMacro           allows usage of eventMacros
-------------------------------------------------------------------------------
```

**Comment by ya4ept** [comment]
Updated documentation: https://openkore.com/wiki/plugin

**Comment by alisonrag** [comment]
Automatic load - ok:
image

Manual load - ok:
image

Manual reload - ok:
image

Manual unload - ok:
image

Manual reload all - ok:
image

Manual unload all - ok:
image

manual load all - ok:
image

**Review by alisonrag — APPROVED** [review]

**Pull request merged**